components/schema-migrator/migrations/director/202102031458_extend_application_resources.up.sql

```diff
@@ -0,0 +1,102 @@
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE bundles
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE api_definitions
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE event_api_definitions
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE documents
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE applications
+    ADD CONSTRAINT application_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE bundles
+    ADD CONSTRAINT bundle_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE api_definitions
+    ADD CONSTRAINT api_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE event_api_definitions
+    ADD CONSTRAINT event_api_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE documents
+    ADD CONSTRAINT document_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE bundles
+    ADD CONSTRAINT bundles_applications_ready_fk
+        FOREIGN KEY (app_id, ready) REFERENCES applications (id, ready) ON UPDATE CASCADE;
+
+ALTER TABLE api_definitions
+    ADD CONSTRAINT api_definitions_bundles_ready_fk
+        FOREIGN KEY (bundle_id, ready) REFERENCES bundles (id, ready) ON UPDATE CASCADE;
+
+ALTER TABLE event_api_definitions
+    ADD CONSTRAINT event_api_definitions_bundles_ready_fk
+        FOREIGN KEY (bundle_id, ready) REFERENCES bundles (id, ready) ON UPDATE CASCADE;
+
+ALTER TABLE documents
+    ADD CONSTRAINT documents_bundles_ready_fk
+        FOREIGN KEY (bundle_id, ready) REFERENCES bundles (id, ready) ON UPDATE CASCADE;
+
+ALTER TYPE webhook_type RENAME TO webhook_type_old;
+CREATE TYPE webhook_type AS ENUM (
+    'CONFIGURATION_CHANGED',
+    'REGISTER_APPLICATION',
+    'DELETE_APPLICATION'
+    );
+ALTER TABLE webhooks ALTER COLUMN type TYPE webhook_type USING type::text::webhook_type;
+DROP TYPE webhook_type_old;
+
+CREATE TYPE webhook_mode AS ENUM (
+    'ASYNC',
+    'SYNC'
+);
+
+ALTER TABLE webhooks
+    ADD COLUMN mode webhook_mode,
+    ADD COLUMN correlation_id_key varchar(256),
+    ADD COLUMN retry_interval int,
+    ADD COLUMN timeout int,
+    ADD COLUMN url_template jsonb,
+    ADD COLUMN input_template jsonb,
```

Comment:
i didnt see any template definitions and validations of templates in the code. They are not arbitary, there are some rules that have to be followed and these rules should come from this component and the validation should be done during webhook creation. We can discuss ideas/proposals.

Reply:
Webhook validation added with: https://github.com/kyma-incubator/compass/pull/1714/commits/e8c3613bd623371576d93844ba5f4326f3c640c2